### PR TITLE
Creating a base job that will debounce duplicates

### PR DESCRIPTION
**Pull request by qschmick**
## Description

Base job that allows for debouncing duplicates. Good to use when a system might dispatch jobs async via different code paths but you only want one instance to launch and the others to be silenced.

## Release Notes

_Suggested details for the release notes to explain how this impacts users_

## Types of changes

- [ ] Bugfix
- [X] New feature
- [ ] Chore (mundane code change)
- [ ] Maintenance (clean up)

## Checklist

- [ ] Unit tests pass locally
- [ ] Added tests to code that was updated or edited
- [ ] Updated/added documentation

### CODE REVIEW

File: src/Jobs/DebouncedJob.php [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    }
+
+    protected function minimumWaitComplete() : bool
+    {
+        /**
+         * @var Carbon|null $minimum
+         */
+        $minimum = Cache::get($this->getMinimumWaitTimeKey());
+
+        return ! $minimum || $minimum->isPast();
```
Comment (solflare) [inline src/Jobs/DebouncedJob.php:104]: Would `! $minimum?->isPast()` work?

**Review by solflare — APPROVED** [review]
🚀 Just one minor question

**Review by jzone1366 — APPROVED** [review]

File: src/Jobs/DebouncedJob.php (hunk 1 above)
Comment (qschmick) [reply src/Jobs/DebouncedJob.php:104]: That would be equal to:

```
! $minimum || ! $minimum->isPast();
```

Comment (solflare) [reply src/Jobs/DebouncedJob.php:104]: Derp I knew I missed something 🤦

**Pull request merged**